File: src/dz3/Main.java
package dz3;


public class Main {
    public static void main() {
        var x = "abcba";
        System.out.println(x + ": " + isPalindrome(x));
        var y = "abc";
        System.out.println(y + ": " + isPalindrome(y));
    }

    private static boolean isPalindrome(String str) {
        var sb = new StringBuilder(str);
        sb.reverse();
        return str.equals(sb.toString());
    }
}

File: src/dz3/Main.kt
package dz3

fun main() {
    val x = "abcba"
    println("$x: ${isPalindrome(x)}")
    val y = "abc"
    println("$y: ${isPalindrome(y)}")
}

fun isPalindrome(str:String): Boolean = str == str.reversed()